boa/src/builtins/map/mod.rs

```diff
@@ -96,14 +98,15 @@ impl Map {
                         map
                     } else if object.is_array() {
                         let mut map = OrderedMap::new();
-                        let len = args[0].get_field("length").to_integer(context)? as i32;
+                        let len = args[0].get_field("length", context)?.to_integer(context)? as i32;
                         for i in 0..len {
-                            let val = &args[0].get_field(i.to_string());
-                            let (key, value) = Self::get_key_value(val).ok_or_else(|| {
-                                context.construct_type_error(
-                                    "iterable for Map should have array-like objects",
-                                )
-                            })?;
+                            let val = &args[0].get_field(i.to_string(), context)?;
```

Comment:
```suggestion
                            let val = &args[0].get_field(i, context)?;
```
I know it was there before, but it's better if we change so we avoid an unnecessary string allocation.